pkg/cmd/check/check.go

```diff
@@ -0,0 +1,126 @@
+package check
+
+import (
+	"fmt"
+	"github.com/jenkins-x-plugins/jx-semanticcheck/pkg/helpers"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/cmdrunner"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/cobras/helper"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/cobras/templates"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/gitclient"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/gitclient/cli"
+	"github.com/jenkins-x/jx-helpers/v3/pkg/options"
+	"github.com/jenkins-x/jx-logging/v3/pkg/log"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+// Options contains the command line flags
+type Options struct {
+	options.BaseOptions
+
+	GitClient     gitclient.Interface
+	CommandRunner cmdrunner.CommandRunner
+
+	Dir string
+}
+
+var (
+	cmdLong = templates.LongDesc(`
+		Checks whether the commit messages in a pull request follow the Conventional Commits specification
+`)
+
+	cmdExample = templates.Examples(`
+		jx-semanticcheck check 
+`)
+
+	ConventionalCommitTypes = []string{"feat", "fix", "perf", "refactor", "docs", "test", "revert", "style", "chore", "build"}
```

Comment:
Perhaps try to use the specification from here https://www.conventionalcommits.org/en/v1.0.0/#specification
The rules that are describe are a little more complex I guess.

Reply:
We discussed this and for now we think this list is ok and will be fine to add to in future.

One of the things that we would like to solve is to allow lighthouse and semanticcheck as well as release updater to all work from the same list and ensure that we're inline without having to completely replicate. But I think we can get this out (as we've already been using this for months) and start putting this into pipelines by default to help guide people to using our releasing features and semantic versioning

Reply:
Fine

Reply:
But where is the list documented?